api/v1beta2/imagepolicy_types.go

```diff
@@ -0,0 +1,174 @@
+	// +required
+	ImageRepositoryRef meta.NamespacedObjectReference `json:"imageRepositoryRef"`
+	// Policy gives the particulars of the policy to be followed in
+	// selecting the most recent image
+	// +required
+	Policy ImagePolicyChoice `json:"policy"`
+	// FilterTags enables filtering for only a subset of tags based on a set of
+	// rules. If no rules are provided, all the tags from the repository will be
+	// ordered and compared.
+	// +optional
+	FilterTags *TagFilter `json:"filterTags,omitempty"`
+}
+
+// ImagePolicyChoice is a union of all the types of policy that can be
+// supplied.
+type ImagePolicyChoice struct {
+	// SemVer gives a semantic version range to check against the tags
+	// available.
+	// +optional
+	SemVer *SemVerPolicy `json:"semver,omitempty"`
+	// Alphabetical set of rules to use for alphabetical ordering of the tags.
+	// +optional
+	Alphabetical *AlphabeticalPolicy `json:"alphabetical,omitempty"`
+	// Numerical set of rules to use for numerical ordering of the tags.
+	// +optional
+	Numerical *NumericalPolicy `json:"numerical,omitempty"`
+}
+
+// SemVerPolicy specifies a semantic version policy.
+type SemVerPolicy struct {
+	// Range gives a semver range for the image tag; the highest
+	// version within the range that's a tag yields the latest image.
+	// +required
+	Range string `json:"range"`
+}
+
+// AlphabeticalPolicy specifies a alphabetical ordering policy.
+type AlphabeticalPolicy struct {
+	// Order specifies the sorting order of the tags. Given the letters of the
+	// alphabet as tags, ascending order would select Z, and descending order
+	// would select A.
+	// +kubebuilder:default:="asc"
+	// +kubebuilder:validation:Enum=asc;desc
+	// +optional
+	Order string `json:"order,omitempty"`
+}
+
+// NumericalPolicy specifies a numerical ordering policy.
+type NumericalPolicy struct {
+	// Order specifies the sorting order of the tags. Given the integer values
+	// from 0 to 9 as tags, ascending order would select 9, and descending order
+	// would select 0.
+	// +kubebuilder:default:="asc"
+	// +kubebuilder:validation:Enum=asc;desc
+	// +optional
+	Order string `json:"order,omitempty"`
+}
+
+// TagFilter enables filtering tags based on a set of defined rules
+type TagFilter struct {
+	// Pattern specifies a regular expression pattern used to filter for image
+	// tags.
+	// +optional
+	Pattern string `json:"pattern"`
+	// Extract allows a capture group to be extracted from the specified regular
+	// expression pattern, useful before tag evaluation.
+	// +optional
+	Extract string `json:"extract"`
+	// Discriminator allows to split the tags in multiple groups and execute the
+	// policy against each of those groups. This value if present is a capture
+	// group to be extracted from the regular expression pattern.
+	// +optional
+	Discriminator string `json:"discriminator,omitempty"`
+	// Additional attributes we want to extract from the tag.
+	// +optional
+	Attributes []string `json:"attributes,omitempty"`
```

Comment:
Thinking on easiest way to use this,

Image automation is a global thing (process multiple policy) so configuration to extract data cannot be set on the image configuration itself 
It may be possible to use additional parameter on the json to apply a policy, but then we may need to duplicate regexp which may be not so easy to use.